src/khoj/routers/indexer.py

```diff
@@ -69,8 +68,8 @@ async def update(
     host: Optional[str] = Header(None),
 ):
     user = request.user.object if request.user.is_authenticated else None
-    if x_api_key != "secret":
-        raise HTTPException(status_code=401, detail="Invalid API Key")
+    if user is None:
+        raise HTTPException(status_code=401, detail="Authenticated user required.")
```

Comment:
Same here, shouldn't be able to reach here with the `authenticated` decorator. In `anonymous-mode`, the only difference is that it uses the `default` user.